osu.Game/Overlays/Settings/SettingsItem.cs

```diff
@@ -137,15 +144,24 @@ private void load()
             {
                 AddInternal(new RestoreDefaultValueButton<T>
                 {
-                    Current = controlWithCurrent.Current,
+                    Current = Current,
                 });
             }
         }
 
         private void updateDisabled()
         {
             if (labelText != null)
-                labelText.Alpha = controlWithCurrent.Current.Disabled ? 0.3f : 1;
+                labelText.Alpha = Current.Disabled ? 0.3f : 1;
+        }
+
+        private class ControlDoesNotImplementCurrentException : InvalidOperationException
+        {
+            public ControlDoesNotImplementCurrentException()
+                : base(@$"Control created via {nameof(CreateControl)} must implement {nameof(IHasCurrentValue<T>)}, "
+                       + @$"or, if custom implementation is intended, override {nameof(Current)} on this item instead.")
```

Comment:
I do not understand this exception message. "custom implementation" of what? You might as well leave it on the first sentence (has to implement `IHasCurrentValue<T>`, period).

Reply:
Custom implementation of how to transfer a "current" bindable to a settings item, I'll leave it at the first sentence, I don't think we'll actually need this at all.